src/advanced_concepts/custom_runners.md

```diff
@@ -1 +1,191 @@
-# Writing custom runners
+# Writing Custom Runners
+
+While the official runners cover most common use cases, you might need to create your own custom runner for specific requirements - like running commands on a remote cluster, implementing custom logging, or handling specialized file systems.
+
+## Understanding the Runner Protocol
+
+Custom runners need to implement two main protocols: `Runner` and `Execution`. Think of the `Runner` as a factory that creates `Execution` objects, and each `Execution` handles a single command run.
+
+### The Runner Protocol
+
+Your runner class needs to implement just one method:
+
+```python
+def start_execution(self, metadata: Metadata) -> Execution:
```

Comment:
Should there be an example of the runner class (e.g. `class ExampleRunner` or something like that?